--- ./harib04e/Makefile	2020-09-08 12:02:04.000000000 +0900
+++ ./harib04d/Makefile	2020-09-08 11:51:00.000000000 +0900
@@ -1,7 +1,6 @@
 TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
-		int.obj fifo.obj
+		int.obj
 
 TOOLPATH = ${TOLSET}/z_tools/
 INCPATH  = ${TOLSET}/z_tools/haribote/
--- ./harib04e/Makefile	2020-09-08 12:02:04.000000000 +0900
+++ ./harib04a/Makefile	2020-09-08 11:59:28.000000000 +0900
@@ -1,7 +1,7 @@
 TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
 
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
-		int.obj fifo.obj
+		int.obj
 
 TOOLPATH = ${TOLSET}/z_tools/
 INCPATH  = ${TOLSET}/z_tools/haribote/
--- ./harib04e/Makefile	2020-09-08 12:02:04.000000000 +0900
+++ ./harib04b/Makefile	2020-09-08 11:51:00.000000000 +0900
@@ -1,7 +1,6 @@
 TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
-		int.obj fifo.obj
+		int.obj
 
 TOOLPATH = ${TOLSET}/z_tools/
 INCPATH  = ${TOLSET}/z_tools/haribote/
--- ./harib04e/Makefile	2020-09-08 12:02:04.000000000 +0900
+++ ./harib04c/Makefile	2020-09-08 11:51:00.000000000 +0900
@@ -1,7 +1,6 @@
 TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
-		int.obj fifo.obj
+		int.obj
 
 TOOLPATH = ${TOLSET}/z_tools/
 INCPATH  = ${TOLSET}/z_tools/haribote/
--- ./harib04e/Makefile	2020-09-08 12:02:04.000000000 +0900
+++ ./z_tools/haribote/Makefile	2020-09-08 11:51:00.000000000 +0900
@@ -1,7 +1,6 @@
 TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
-		int.obj fifo.obj
+		int.obj
 
 TOOLPATH = ${TOLSET}/z_tools/
 INCPATH  = ${TOLSET}/z_tools/haribote/
