--- ./harib22a/Makefile	2005-11-12 13:31:04.000000000 +0900
+++ ./harib22b/Makefile	2005-11-12 20:42:08.000000000 +0900
@@ -150,10 +150,17 @@
 beepdown.hrb : beepdown.bim Makefile
 	$(BIM2HRB) beepdown.bim beepdown.hrb 40k
 
+color.bim : color.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color.bim stack:1k map:color.map \
+		color.obj a_nask.obj
+
+color.hrb : color.bim Makefile
+	$(BIM2HRB) color.bim color.hrb 56k
+
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
 		winhelo.hrb winhelo2.hrb winhelo3.hrb star1.hrb stars.hrb stars2.hrb \
-		lines.hrb walk.hrb noodle.hrb beepdown.hrb
+		lines.hrb walk.hrb noodle.hrb beepdown.hrb color.hrb
 	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -175,6 +182,7 @@
 		copy from:walk.hrb to:@: \
 		copy from:noodle.hrb to:@: \
 		copy from:beepdown.hrb to:@: \
+		copy from:color.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib22a/Makefile	2005-11-12 13:31:04.000000000 +0900
+++ ./harib22f/Makefile	2006-01-29 23:49:44.000000000 +0900
@@ -150,10 +150,24 @@
 beepdown.hrb : beepdown.bim Makefile
 	$(BIM2HRB) beepdown.bim beepdown.hrb 40k
 
+color.bim : color.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color.bim stack:1k map:color.map \
+		color.obj a_nask.obj
+
+color.hrb : color.bim Makefile
+	$(BIM2HRB) color.bim color.hrb 56k
+
+color2.bim : color2.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color2.bim stack:1k map:color2.map \
+		color2.obj a_nask.obj
+
+color2.hrb : color2.bim Makefile
+	$(BIM2HRB) color2.bim color2.hrb 56k
+
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
 		winhelo.hrb winhelo2.hrb winhelo3.hrb star1.hrb stars.hrb stars2.hrb \
-		lines.hrb walk.hrb noodle.hrb beepdown.hrb
+		lines.hrb walk.hrb noodle.hrb beepdown.hrb color.hrb color2.hrb
 	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -175,6 +189,8 @@
 		copy from:walk.hrb to:@: \
 		copy from:noodle.hrb to:@: \
 		copy from:beepdown.hrb to:@: \
+		copy from:color.hrb to:@: \
+		copy from:color2.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib22a/Makefile	2005-11-12 13:31:04.000000000 +0900
+++ ./harib22e/Makefile	2006-01-29 23:49:14.000000000 +0900
@@ -150,10 +150,24 @@
 beepdown.hrb : beepdown.bim Makefile
 	$(BIM2HRB) beepdown.bim beepdown.hrb 40k
 
+color.bim : color.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color.bim stack:1k map:color.map \
+		color.obj a_nask.obj
+
+color.hrb : color.bim Makefile
+	$(BIM2HRB) color.bim color.hrb 56k
+
+color2.bim : color2.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color2.bim stack:1k map:color2.map \
+		color2.obj a_nask.obj
+
+color2.hrb : color2.bim Makefile
+	$(BIM2HRB) color2.bim color2.hrb 56k
+
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
 		winhelo.hrb winhelo2.hrb winhelo3.hrb star1.hrb stars.hrb stars2.hrb \
-		lines.hrb walk.hrb noodle.hrb beepdown.hrb
+		lines.hrb walk.hrb noodle.hrb beepdown.hrb color.hrb color2.hrb
 	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -175,6 +189,8 @@
 		copy from:walk.hrb to:@: \
 		copy from:noodle.hrb to:@: \
 		copy from:beepdown.hrb to:@: \
+		copy from:color.hrb to:@: \
+		copy from:color2.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib22a/Makefile	2005-11-12 13:31:04.000000000 +0900
+++ ./harib22d/Makefile	2006-01-29 23:48:42.000000000 +0900
@@ -150,10 +150,24 @@
 beepdown.hrb : beepdown.bim Makefile
 	$(BIM2HRB) beepdown.bim beepdown.hrb 40k
 
+color.bim : color.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color.bim stack:1k map:color.map \
+		color.obj a_nask.obj
+
+color.hrb : color.bim Makefile
+	$(BIM2HRB) color.bim color.hrb 56k
+
+color2.bim : color2.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color2.bim stack:1k map:color2.map \
+		color2.obj a_nask.obj
+
+color2.hrb : color2.bim Makefile
+	$(BIM2HRB) color2.bim color2.hrb 56k
+
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
 		winhelo.hrb winhelo2.hrb winhelo3.hrb star1.hrb stars.hrb stars2.hrb \
-		lines.hrb walk.hrb noodle.hrb beepdown.hrb
+		lines.hrb walk.hrb noodle.hrb beepdown.hrb color.hrb color2.hrb
 	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -175,6 +189,8 @@
 		copy from:walk.hrb to:@: \
 		copy from:noodle.hrb to:@: \
 		copy from:beepdown.hrb to:@: \
+		copy from:color.hrb to:@: \
+		copy from:color2.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib22a/Makefile	2005-11-12 13:31:04.000000000 +0900
+++ ./harib22g/Makefile	2006-01-29 23:50:24.000000000 +0900
@@ -150,10 +150,24 @@
 beepdown.hrb : beepdown.bim Makefile
 	$(BIM2HRB) beepdown.bim beepdown.hrb 40k
 
+color.bim : color.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color.bim stack:1k map:color.map \
+		color.obj a_nask.obj
+
+color.hrb : color.bim Makefile
+	$(BIM2HRB) color.bim color.hrb 56k
+
+color2.bim : color2.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color2.bim stack:1k map:color2.map \
+		color2.obj a_nask.obj
+
+color2.hrb : color2.bim Makefile
+	$(BIM2HRB) color2.bim color2.hrb 56k
+
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
 		winhelo.hrb winhelo2.hrb winhelo3.hrb star1.hrb stars.hrb stars2.hrb \
-		lines.hrb walk.hrb noodle.hrb beepdown.hrb
+		lines.hrb walk.hrb noodle.hrb beepdown.hrb color.hrb color2.hrb
 	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -175,6 +189,8 @@
 		copy from:walk.hrb to:@: \
 		copy from:noodle.hrb to:@: \
 		copy from:beepdown.hrb to:@: \
+		copy from:color.hrb to:@: \
+		copy from:color2.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib22a/Makefile	2005-11-12 13:31:04.000000000 +0900
+++ ./harib22j/Makefile	2006-01-29 23:56:20.000000000 +0900
@@ -150,10 +150,24 @@
 beepdown.hrb : beepdown.bim Makefile
 	$(BIM2HRB) beepdown.bim beepdown.hrb 40k
 
+color.bim : color.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color.bim stack:1k map:color.map \
+		color.obj a_nask.obj
+
+color.hrb : color.bim Makefile
+	$(BIM2HRB) color.bim color.hrb 56k
+
+color2.bim : color2.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color2.bim stack:1k map:color2.map \
+		color2.obj a_nask.obj
+
+color2.hrb : color2.bim Makefile
+	$(BIM2HRB) color2.bim color2.hrb 56k
+
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
 		winhelo.hrb winhelo2.hrb winhelo3.hrb star1.hrb stars.hrb stars2.hrb \
-		lines.hrb walk.hrb noodle.hrb beepdown.hrb
+		lines.hrb walk.hrb noodle.hrb beepdown.hrb color.hrb color2.hrb
 	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -175,6 +189,8 @@
 		copy from:walk.hrb to:@: \
 		copy from:noodle.hrb to:@: \
 		copy from:beepdown.hrb to:@: \
+		copy from:color.hrb to:@: \
+		copy from:color2.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib22a/Makefile	2005-11-12 13:31:04.000000000 +0900
+++ ./harib22i/Makefile	2006-01-29 23:52:46.000000000 +0900
@@ -150,10 +150,24 @@
 beepdown.hrb : beepdown.bim Makefile
 	$(BIM2HRB) beepdown.bim beepdown.hrb 40k
 
+color.bim : color.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color.bim stack:1k map:color.map \
+		color.obj a_nask.obj
+
+color.hrb : color.bim Makefile
+	$(BIM2HRB) color.bim color.hrb 56k
+
+color2.bim : color2.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color2.bim stack:1k map:color2.map \
+		color2.obj a_nask.obj
+
+color2.hrb : color2.bim Makefile
+	$(BIM2HRB) color2.bim color2.hrb 56k
+
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
 		winhelo.hrb winhelo2.hrb winhelo3.hrb star1.hrb stars.hrb stars2.hrb \
-		lines.hrb walk.hrb noodle.hrb beepdown.hrb
+		lines.hrb walk.hrb noodle.hrb beepdown.hrb color.hrb color2.hrb
 	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -175,6 +189,8 @@
 		copy from:walk.hrb to:@: \
 		copy from:noodle.hrb to:@: \
 		copy from:beepdown.hrb to:@: \
+		copy from:color.hrb to:@: \
+		copy from:color2.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib22a/Makefile	2005-11-12 13:31:04.000000000 +0900
+++ ./harib22h/Makefile	2006-01-29 23:50:52.000000000 +0900
@@ -150,10 +150,24 @@
 beepdown.hrb : beepdown.bim Makefile
 	$(BIM2HRB) beepdown.bim beepdown.hrb 40k
 
+color.bim : color.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color.bim stack:1k map:color.map \
+		color.obj a_nask.obj
+
+color.hrb : color.bim Makefile
+	$(BIM2HRB) color.bim color.hrb 56k
+
+color2.bim : color2.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color2.bim stack:1k map:color2.map \
+		color2.obj a_nask.obj
+
+color2.hrb : color2.bim Makefile
+	$(BIM2HRB) color2.bim color2.hrb 56k
+
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
 		winhelo.hrb winhelo2.hrb winhelo3.hrb star1.hrb stars.hrb stars2.hrb \
-		lines.hrb walk.hrb noodle.hrb beepdown.hrb
+		lines.hrb walk.hrb noodle.hrb beepdown.hrb color.hrb color2.hrb
 	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -175,6 +189,8 @@
 		copy from:walk.hrb to:@: \
 		copy from:noodle.hrb to:@: \
 		copy from:beepdown.hrb to:@: \
+		copy from:color.hrb to:@: \
+		copy from:color2.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib22a/Makefile	2005-11-12 13:31:04.000000000 +0900
+++ ./harib22c/Makefile	2006-01-29 23:48:04.000000000 +0900
@@ -150,10 +150,24 @@
 beepdown.hrb : beepdown.bim Makefile
 	$(BIM2HRB) beepdown.bim beepdown.hrb 40k
 
+color.bim : color.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color.bim stack:1k map:color.map \
+		color.obj a_nask.obj
+
+color.hrb : color.bim Makefile
+	$(BIM2HRB) color.bim color.hrb 56k
+
+color2.bim : color2.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:color2.bim stack:1k map:color2.map \
+		color2.obj a_nask.obj
+
+color2.hrb : color2.bim Makefile
+	$(BIM2HRB) color2.bim color2.hrb 56k
+
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
 		winhelo.hrb winhelo2.hrb winhelo3.hrb star1.hrb stars.hrb stars2.hrb \
-		lines.hrb walk.hrb noodle.hrb beepdown.hrb
+		lines.hrb walk.hrb noodle.hrb beepdown.hrb color.hrb color2.hrb
 	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -175,6 +189,8 @@
 		copy from:walk.hrb to:@: \
 		copy from:noodle.hrb to:@: \
 		copy from:beepdown.hrb to:@: \
+		copy from:color.hrb to:@: \
+		copy from:color2.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
