--- ../16_day/harib13e/Makefile	2020-11-19 05:58:54.736648407 +0900
+++ ./harib14e/Makefile	2005-06-11 00:48:16.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -53,7 +50,7 @@
 	copy /B asmhead.bin+bootpack.hrb haribote.sys
 
 haribote.img : ipl10.bin haribote.sys Makefile
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		imgout:haribote.img
@@ -76,8 +73,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
--- ../16_day/harib13e/Makefile	2020-11-19 05:58:54.736648407 +0900
+++ ./harib14g/Makefile	2005-06-11 00:48:16.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -53,7 +50,7 @@
 	copy /B asmhead.bin+bootpack.hrb haribote.sys
 
 haribote.img : ipl10.bin haribote.sys Makefile
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		imgout:haribote.img
@@ -76,8 +73,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
--- ../16_day/harib13e/Makefile	2020-11-19 05:58:54.736648407 +0900
+++ ./harib14a/Makefile	2005-06-11 00:48:16.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -53,7 +50,7 @@
 	copy /B asmhead.bin+bootpack.hrb haribote.sys
 
 haribote.img : ipl10.bin haribote.sys Makefile
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		imgout:haribote.img
@@ -76,8 +73,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
--- ../16_day/harib13e/Makefile	2020-11-19 05:58:54.736648407 +0900
+++ ./harib14d/Makefile	2005-06-11 00:48:16.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -53,7 +50,7 @@
 	copy /B asmhead.bin+bootpack.hrb haribote.sys
 
 haribote.img : ipl10.bin haribote.sys Makefile
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		imgout:haribote.img
@@ -76,8 +73,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
--- ../16_day/harib13e/Makefile	2020-11-19 05:58:54.736648407 +0900
+++ ./harib14b/Makefile	2005-06-11 00:48:16.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -53,7 +50,7 @@
 	copy /B asmhead.bin+bootpack.hrb haribote.sys
 
 haribote.img : ipl10.bin haribote.sys Makefile
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		imgout:haribote.img
@@ -76,8 +73,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
--- ../16_day/harib13e/Makefile	2020-11-19 05:58:54.736648407 +0900
+++ ./harib14c/Makefile	2005-06-11 00:48:16.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -53,7 +50,7 @@
 	copy /B asmhead.bin+bootpack.hrb haribote.sys
 
 haribote.img : ipl10.bin haribote.sys Makefile
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		imgout:haribote.img
@@ -76,8 +73,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
--- ../16_day/harib13e/Makefile	2020-11-19 05:58:54.736648407 +0900
+++ ./harib14f/Makefile	2005-06-11 00:48:16.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -53,7 +50,7 @@
 	copy /B asmhead.bin+bootpack.hrb haribote.sys
 
 haribote.img : ipl10.bin haribote.sys Makefile
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		imgout:haribote.img
@@ -76,8 +73,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
