--- ./harib19d/Makefile	2021-03-17 01:57:36.994478541 +0900
+++ ./harib19b/Makefile	2021-03-17 01:38:06.174090868 +0900
@@ -70,34 +70,14 @@
 hello3.hrb : hello3.bim Makefile
 	$(BIM2HRB) hello3.bim hello3.hrb 0
 
-hello4.bim : hello4.obj a_nask.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:hello4.bim stack:1k map:hello4.map \
-		hello4.obj a_nask.obj
-
-hello4.hrb : hello4.bim Makefile
-	$(BIM2HRB) hello4.bim hello4.hrb 0
-
 bug1.bim : bug1.obj Makefile
 	$(OBJ2BIM) @$(RULEFILE) out:bug1.bim map:bug1.map bug1.obj a_nask.obj
 
 bug1.hrb : bug1.bim Makefile
 	$(BIM2HRB) bug1.bim bug1.hrb 0
 
-bug2.bim : bug2.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:bug2.bim map:bug2.map bug2.obj
-
-bug2.hrb : bug2.bim Makefile
-	$(BIM2HRB) bug2.bim bug2.hrb 0
-
-bug3.bim : bug3.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:bug3.bim map:bug3.map bug3.obj a_nask.obj
-
-bug3.hrb : bug3.bim Makefile
-	$(BIM2HRB) bug3.bim bug3.hrb 0
-
 haribote.img : ipl10.bin haribote.sys Makefile \
-		hello.hrb hello2.hrb a.hrb hello3.hrb bug1.hrb bug2.hrb bug3.hrb \
-		hello4.hrb
+		hello.hrb hello2.hrb a.hrb hello3.hrb bug1.hrb
 	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -108,9 +88,6 @@
 		copy from:a.hrb to:@: \
 		copy from:hello3.hrb to:@: \
 		copy from:bug1.hrb to:@: \
-		copy from:bug2.hrb to:@: \
-		copy from:bug3.hrb to:@: \
-		copy from:hello4.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib19d/Makefile	2021-03-17 01:57:36.994478541 +0900
+++ ./harib19e/Makefile	2005-10-18 01:55:58.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj window.obj console.obj file.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -77,6 +74,12 @@
 hello4.hrb : hello4.bim Makefile
 	$(BIM2HRB) hello4.bim hello4.hrb 0
 
+hello5.bim : hello5.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:hello5.bim stack:1k map:hello5.map hello5.obj
+
+hello5.hrb : hello5.bim Makefile
+	$(BIM2HRB) hello5.bim hello5.hrb 0
+
 bug1.bim : bug1.obj Makefile
 	$(OBJ2BIM) @$(RULEFILE) out:bug1.bim map:bug1.map bug1.obj a_nask.obj
 
@@ -97,12 +100,12 @@
 
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb bug1.hrb bug2.hrb bug3.hrb \
-		hello4.hrb
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+		hello4.hrb hello5.hrb
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		copy from:ipl10.nas to:@: \
-		copy from:keyboard.c to:@: \
+		copy from:make.bat to:@: \
 		copy from:hello.hrb to:@: \
 		copy from:hello2.hrb to:@: \
 		copy from:a.hrb to:@: \
@@ -111,6 +114,7 @@
 		copy from:bug2.hrb to:@: \
 		copy from:bug3.hrb to:@: \
 		copy from:hello4.hrb to:@: \
+		copy from:hello5.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
@@ -131,8 +135,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
--- ./harib19d/Makefile	2021-03-17 01:57:36.994478541 +0900
+++ ./harib19c/Makefile	2021-03-17 01:51:36.520397129 +0900
@@ -70,13 +70,6 @@
 hello3.hrb : hello3.bim Makefile
 	$(BIM2HRB) hello3.bim hello3.hrb 0
 
-hello4.bim : hello4.obj a_nask.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:hello4.bim stack:1k map:hello4.map \
-		hello4.obj a_nask.obj
-
-hello4.hrb : hello4.bim Makefile
-	$(BIM2HRB) hello4.bim hello4.hrb 0
-
 bug1.bim : bug1.obj Makefile
 	$(OBJ2BIM) @$(RULEFILE) out:bug1.bim map:bug1.map bug1.obj a_nask.obj
 
@@ -96,8 +89,7 @@
 	$(BIM2HRB) bug3.bim bug3.hrb 0
 
 haribote.img : ipl10.bin haribote.sys Makefile \
-		hello.hrb hello2.hrb a.hrb hello3.hrb bug1.hrb bug2.hrb bug3.hrb \
-		hello4.hrb
+		hello.hrb hello2.hrb a.hrb hello3.hrb bug1.hrb bug2.hrb bug3.hrb
 	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -110,7 +102,6 @@
 		copy from:bug1.hrb to:@: \
 		copy from:bug2.hrb to:@: \
 		copy from:bug3.hrb to:@: \
-		copy from:hello4.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib19d/Makefile	2021-03-17 01:57:36.994478541 +0900
+++ ./harib19f/Makefile	2005-10-18 23:49:54.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj window.obj console.obj file.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -77,6 +74,19 @@
 hello4.hrb : hello4.bim Makefile
 	$(BIM2HRB) hello4.bim hello4.hrb 0
 
+hello5.bim : hello5.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:hello5.bim stack:1k map:hello5.map hello5.obj
+
+hello5.hrb : hello5.bim Makefile
+	$(BIM2HRB) hello5.bim hello5.hrb 0
+
+winhelo.bim : winhelo.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:winhelo.bim stack:1k map:winhelo.map \
+		winhelo.obj a_nask.obj
+
+winhelo.hrb : winhelo.bim Makefile
+	$(BIM2HRB) winhelo.bim winhelo.hrb 0
+
 bug1.bim : bug1.obj Makefile
 	$(OBJ2BIM) @$(RULEFILE) out:bug1.bim map:bug1.map bug1.obj a_nask.obj
 
@@ -97,12 +107,12 @@
 
 haribote.img : ipl10.bin haribote.sys Makefile \
 		hello.hrb hello2.hrb a.hrb hello3.hrb bug1.hrb bug2.hrb bug3.hrb \
-		hello4.hrb
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+		hello4.hrb hello5.hrb winhelo.hrb
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		copy from:ipl10.nas to:@: \
-		copy from:keyboard.c to:@: \
+		copy from:make.bat to:@: \
 		copy from:hello.hrb to:@: \
 		copy from:hello2.hrb to:@: \
 		copy from:a.hrb to:@: \
@@ -111,6 +121,8 @@
 		copy from:bug2.hrb to:@: \
 		copy from:bug3.hrb to:@: \
 		copy from:hello4.hrb to:@: \
+		copy from:hello5.hrb to:@: \
+		copy from:winhelo.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
@@ -131,8 +143,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
--- ./harib19d/Makefile	2021-03-17 01:57:36.994478541 +0900
+++ ./harib19a/Makefile	2021-03-17 01:26:59.907101521 +0900
@@ -70,34 +70,30 @@
 hello3.hrb : hello3.bim Makefile
 	$(BIM2HRB) hello3.bim hello3.hrb 0
 
-hello4.bim : hello4.obj a_nask.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:hello4.bim stack:1k map:hello4.map \
-		hello4.obj a_nask.obj
+crack1.bim : crack1.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:crack1.bim map:crack1.map crack1.obj a_nask.obj
 
-hello4.hrb : hello4.bim Makefile
-	$(BIM2HRB) hello4.bim hello4.hrb 0
+crack1.hrb : crack1.bim Makefile
+	$(BIM2HRB) crack1.bim crack1.hrb 0
 
-bug1.bim : bug1.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:bug1.bim map:bug1.map bug1.obj a_nask.obj
+crack2.hrb : crack2.nas Makefile
+	$(NASK) crack2.nas crack2.hrb crack2.lst
 
-bug1.hrb : bug1.bim Makefile
-	$(BIM2HRB) bug1.bim bug1.hrb 0
+crack3.hrb : crack3.nas Makefile
+	$(NASK) crack3.nas crack3.hrb crack3.lst
 
-bug2.bim : bug2.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:bug2.bim map:bug2.map bug2.obj
+crack4.hrb : crack4.nas Makefile
+	$(NASK) crack4.nas crack4.hrb crack4.lst
 
-bug2.hrb : bug2.bim Makefile
-	$(BIM2HRB) bug2.bim bug2.hrb 0
+crack5.hrb : crack5.nas Makefile
+	$(NASK) crack5.nas crack5.hrb crack5.lst
 
-bug3.bim : bug3.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:bug3.bim map:bug3.map bug3.obj a_nask.obj
-
-bug3.hrb : bug3.bim Makefile
-	$(BIM2HRB) bug3.bim bug3.hrb 0
+crack6.hrb : crack6.nas Makefile
+	$(NASK) crack6.nas crack6.hrb crack6.lst
 
 haribote.img : ipl10.bin haribote.sys Makefile \
-		hello.hrb hello2.hrb a.hrb hello3.hrb bug1.hrb bug2.hrb bug3.hrb \
-		hello4.hrb
+		hello.hrb hello2.hrb a.hrb hello3.hrb crack1.hrb crack2.hrb crack3.hrb \
+		crack4.hrb crack5.hrb crack6.hrb
 	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
@@ -107,10 +103,12 @@
 		copy from:hello2.hrb to:@: \
 		copy from:a.hrb to:@: \
 		copy from:hello3.hrb to:@: \
-		copy from:bug1.hrb to:@: \
-		copy from:bug2.hrb to:@: \
-		copy from:bug3.hrb to:@: \
-		copy from:hello4.hrb to:@: \
+		copy from:crack1.hrb to:@: \
+		copy from:crack2.hrb to:@: \
+		copy from:crack3.hrb to:@: \
+		copy from:crack4.hrb to:@: \
+		copy from:crack5.hrb to:@: \
+		copy from:crack6.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
--- ./harib19d/Makefile	2021-03-17 01:57:36.994478541 +0900
+++ ./harib19g/Makefile	2005-10-19 15:08:48.000000000 +0900
@@ -1,11 +1,9 @@
-TOLSET     = F:\TEMP\hariboteos\hariboteos\tolset
-
 OBJS_BOOTPACK = bootpack.obj naskfunc.obj hankaku.obj graphic.obj dsctbl.obj \
 		int.obj fifo.obj keyboard.obj mouse.obj memory.obj sheet.obj timer.obj \
 		mtask.obj window.obj console.obj file.obj
 
-TOOLPATH = ${TOLSET}/z_tools/
-INCPATH  = ${TOLSET}/z_tools/haribote/
+TOOLPATH = ../z_tools/
+INCPATH  = ../z_tools/haribote/
 
 MAKE     = $(TOOLPATH)make.exe -r
 NASK     = $(TOOLPATH)nask.exe
@@ -41,7 +39,6 @@
 	$(BIN2OBJ) hankaku.bin hankaku.obj _hankaku
 
 bootpack.bim : $(OBJS_BOOTPACK) Makefile
-	XCOPY ${TOLSET}\z_tools\haribote ..\z_tools\haribote /I /Y 
 	$(OBJ2BIM) @$(RULEFILE) out:bootpack.bim stack:3136k map:bootpack.map \
 		$(OBJS_BOOTPACK)
 # 3MB+64KB=3136KB
@@ -77,40 +74,42 @@
 hello4.hrb : hello4.bim Makefile
 	$(BIM2HRB) hello4.bim hello4.hrb 0
 
-bug1.bim : bug1.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:bug1.bim map:bug1.map bug1.obj a_nask.obj
+hello5.bim : hello5.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:hello5.bim stack:1k map:hello5.map hello5.obj
 
-bug1.hrb : bug1.bim Makefile
-	$(BIM2HRB) bug1.bim bug1.hrb 0
+hello5.hrb : hello5.bim Makefile
+	$(BIM2HRB) hello5.bim hello5.hrb 0
 
-bug2.bim : bug2.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:bug2.bim map:bug2.map bug2.obj
+winhelo.bim : winhelo.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:winhelo.bim stack:1k map:winhelo.map \
+		winhelo.obj a_nask.obj
 
-bug2.hrb : bug2.bim Makefile
-	$(BIM2HRB) bug2.bim bug2.hrb 0
+winhelo.hrb : winhelo.bim Makefile
+	$(BIM2HRB) winhelo.bim winhelo.hrb 0
 
-bug3.bim : bug3.obj Makefile
-	$(OBJ2BIM) @$(RULEFILE) out:bug3.bim map:bug3.map bug3.obj a_nask.obj
+winhelo2.bim : winhelo2.obj a_nask.obj Makefile
+	$(OBJ2BIM) @$(RULEFILE) out:winhelo2.bim stack:1k map:winhelo2.map \
+		winhelo2.obj a_nask.obj
 
-bug3.hrb : bug3.bim Makefile
-	$(BIM2HRB) bug3.bim bug3.hrb 0
+winhelo2.hrb : winhelo2.bim Makefile
+	$(BIM2HRB) winhelo2.bim winhelo2.hrb 0
 
 haribote.img : ipl10.bin haribote.sys Makefile \
-		hello.hrb hello2.hrb a.hrb hello3.hrb bug1.hrb bug2.hrb bug3.hrb \
-		hello4.hrb
-	$(EDIMG)   imgin:${TOLSET}/z_tools/fdimg0at.tek \
+		hello.hrb hello2.hrb a.hrb hello3.hrb hello4.hrb hello5.hrb \
+		winhelo.hrb winhelo2.hrb
+	$(EDIMG)   imgin:../z_tools/fdimg0at.tek \
 		wbinimg src:ipl10.bin len:512 from:0 to:0 \
 		copy from:haribote.sys to:@: \
 		copy from:ipl10.nas to:@: \
-		copy from:keyboard.c to:@: \
+		copy from:make.bat to:@: \
 		copy from:hello.hrb to:@: \
 		copy from:hello2.hrb to:@: \
 		copy from:a.hrb to:@: \
 		copy from:hello3.hrb to:@: \
-		copy from:bug1.hrb to:@: \
-		copy from:bug2.hrb to:@: \
-		copy from:bug3.hrb to:@: \
 		copy from:hello4.hrb to:@: \
+		copy from:hello5.hrb to:@: \
+		copy from:winhelo.hrb to:@: \
+		copy from:winhelo2.hrb to:@: \
 		imgout:haribote.img
 
 # 一般規則
@@ -131,8 +130,8 @@
 
 run :
 	$(MAKE) img
-	$(COPY) haribote.img ${TOLSET}\z_tools\qemu\fdimage0.bin
-	$(MAKE) -C ${TOLSET}/z_tools/qemu
+	$(COPY) haribote.img ..\z_tools\qemu\fdimage0.bin
+	$(MAKE) -C ../z_tools/qemu
 
 install :
 	$(MAKE) img
